packages/react-ai/src/components/AIConversation/views/Controls/InputControl.tsx

```diff
@@ -0,0 +1,222 @@
+  className: `${INPUT_BLOCK}__attach-file-input`,
+});
+
+const AttachFileInput: typeof AttachFileInputBase = React.forwardRef(
+  function AttachFileInput(props, ref) {
+    // TODO should attach file to message
+    return <AttachFileInputBase {...props} ref={ref} />;
+  }
+);
+
+const AttachFileControl: AttachFileControl = () => {
+  return (
+    <AttachFileContainer>
+      <AttachFileButton>
+        <AttachIcon />
+      </AttachFileButton>
+      <VisuallyHidden>
+        <AttachFileInput />
+      </VisuallyHidden>
+    </AttachFileContainer>
+  );
+};
+
+AttachFileControl.Container = AttachFileContainer;
+AttachFileControl.Button = AttachFileButton;
+AttachFileControl.Input = AttachFileInput;
+AttachFileControl.VisuallyHidden = VisuallyHidden;
+
+interface AttachFileControl<
+  T extends Partial<AIConversationElements> = AIConversationElements,
+> {
+  (): React.JSX.Element;
+  Button: T['Button'];
+  Container: T['View'];
+  Input: T['Input'];
+  VisuallyHidden: T['View'];
+}
+
+const sendIconProps = () => ({
+  children: (
+    <svg>
+      <g clipPath="url(#clip0_387_1426)">
+        <path
+          d="M14.3199 7.98214L1.66669 1.23486L3.97674 7.98214M14.3199 7.98214L1.66669 14.5689L3.97674 7.98214M14.3199 7.98214H3.97674"
+          strokeWidth="1.5"
+          strokeLinecap="square"
+          strokeLinejoin="round"
+        />
+      </g>
+      <defs>
+        <clipPath id="clip0_387_1426">
+          <rect width="100%" height="100%" />
+        </clipPath>
+      </defs>
+    </svg>
+  ),
+  stroke: 'white',
+  ...iconAttributes,
+});
+
+const SendIcon = withBaseElementProps(Icon, sendIconProps);
+
+const SendButtonBase = withBaseElementProps(Button, {
+  'aria-label': 'Send message',
+  className: `${INPUT_BLOCK}__button ${INPUT_BLOCK}__button--send`,
+  disabled: false,
+  'aria-disabled': false,
+});
+
+const SendButton: typeof SendButtonBase = React.forwardRef(
+  function SendButton(props, ref) {
+    // TODO should come from context
+    const isWaitingForResponse = false;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- TODO send message on click
+    const { input } = React.useContext(InputContext);
+    return (
+      <SendButtonBase
+        onClick={() => {}}
+        disabled={isWaitingForResponse}
+        aria-disabled={isWaitingForResponse}
+        {...props}
+        ref={ref}
+      />
+    );
+  }
+);
+
+const TextInputBase = withBaseElementProps(TextArea, {
+  className: `${INPUT_BLOCK}__input`,
+  placeholder: 'Message Raven',
+  onChange: () => {},
+  id: `${INPUT_BLOCK}-text-input`,
+});
+
+const TextInput: typeof TextInputBase = React.forwardRef(
+  function TextInput(props, ref) {
+    const { input, setInput } = React.useContext(InputContext);
+    // TODO should come from context or prop
+    const isFirstMessage = false;
+
+    React.useEffect(() => {
+      const textarea = document.getElementById(`${INPUT_BLOCK}-text-input`);
+
+      const handleResize = () => {
```

Comment:
https://www.geeksforgeeks.org/how-to-create-auto-resize-textarea-using-javascript-jquery/